wnfs/src/private/file.rs

```diff
@@ -504,17 +504,66 @@ impl PrivateFile {
         }
     }
 
-    /// Reads a number of bytes starting from a given offset.
+    /// Read the contents of this file.
+    /// You can provide a byte offset from which to start reading,
+    /// and you can provide a maximum amount of bytes you want to read.
+    ///
+    /// For more advanced cases, consider using `stream_content` instead.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use anyhow::Result;
+    /// use chrono::Utc;
+    /// use rand_chacha::ChaCha12Rng;
+    /// use rand_core::SeedableRng;
+    /// use wnfs::{
+    ///     private::{PrivateDirectory, PrivateFile, forest::{hamt::HamtForest, traits::PrivateForest}},
+    ///     common::{MemoryBlockStore, utils::get_random_bytes},
+    /// };
+    ///
+    /// #[async_std::main]
+    /// async fn main() -> Result<()> {
+    ///     let store = &MemoryBlockStore::new();
+    ///     let rng = &mut ChaCha12Rng::from_entropy();
+    ///     let forest = &mut HamtForest::new_rsa_2048_rc(rng);
+    ///     let content = b"Hello, World!\n".repeat(1000).to_vec();
+    ///
+    ///     let file = PrivateFile::with_content(
+    ///         &forest.empty_name(),
+    ///         Utc::now(),
+    ///         content,
+    ///         forest,
+    ///         store,
+    ///         rng,
+    ///     )
+    ///     .await
+    ///     .unwrap();
+    ///
+    ///     let content = file.read_at(14, Some(28), forest, store).await?;
+    ///
+    ///     assert_eq!(content, b"Hello, World!\nHello, World!\n");
+    ///
+    ///     Ok(())
+    /// }
+    /// ```
     pub async fn read_at<'a>(
         &'a self,
-        offset: usize,
-        len: usize,
+        byte_offset: usize,
```

Comment:
While you're here, is it fine if I ask you to change `byte_offset: usize` to `byte_offset: u64`, similar to what's in the public side?
It's a tiny bit of a refactor. The `u64` should carry through to the `.stream` call (`block_index` should also be a `u64`), the very leaf of the call tree being `create_block_name`, which today turns the `usize` block index into a `u64`, but should instead just take a `u64`.

(And `len`s and `len_limit`s generally should keep being `usize`s)

Reply:
Sure thing. One thing that's a bit annoying here is that a lot of the other numbers we're working with are `usize`s. Do we cast those usizes to u64, or the other way around?

Reply:
for block indices, work in u64 for as long as possible. We should only need to cast that to `usize` to work with native array indices.

Reply:
Ok, updated. Not sure if this was the right approach, feels a bit messy.

Reply:
It's okay. The most important piece is getting the API to support very large files & to align public and private APIs.
We can do a cleanup round later :v: